### DISCUSSION TITLE
Possible to repaint the prompt right before command exec?

### DISCUSSION THREAD

**Discussion by yump** [fish-shell/fish-shell#10930] [Q&A] [answered]
Some of the prompts have clocks in them, which is theoretically useful to see how long a command has been running for. However, because of the way `fish_prompt` works, the time shown is actually the time the last command finished, not the time the currently running command started.

Another use for this would be the VCS prompt.  Consider this situation commonly encountered when writing scripts

1. Terminal in clean working tree: `> ./something.py`.  
2. Modify and save `something.py` in an open editor.  
3. Same terminal: `> ./something.py`

The second execution of `something.py` will appear as if it happened in a clean tree, when it in fact did not. 

Things I have tried so far in config.fish (version 3.7.0 at time of writing) that haven't worked:

```
function on_preexec_refresh_prompt --on-event fish_preexec
    commandline -f repaint
end
```
```
bind -k enter 'commandline -f repaint'
```

**Comment by krobelus** [comment, marked as answer]
> bind -k enter 'commandline -f repaint'

if I do that I get

```
$ bind -k enter
bind: Key with name 'enter' does not have any mapping
```

This works for me: `bind \r repaint execute`
next release supports `bind enter repaint execute`

**Reply by yump** [reply]
`bind \r repaint execute` worked great, thanks.
